components/ShowDatesList/index.js

```diff
@@ -0,0 +1,131 @@
+import moment from "moment";
+import { useState } from "react";
+import {
+  StyeldDatesList,
+  StyledDatesDetailCard,
+  StyledDatesDetailCardButton,
+  StyledDatesListRow,
+} from "./styled";
+
+export default function ShowDateList({ dates, places }) {
+  const sortDates = [...dates]
+    .filter(
+      (date) =>
+        moment(date.data.date).format(`YYYY-MM-DD`) >=
+        moment().format(`YYYY-MM-DD`)
+    )
+    .sort((a, b) => new Date(a.data.date) - new Date(b.data.date));
+
+  const [clickindex, setClickIndex] = useState(null);
+
+  function handelClickDate(index) {
+    setClickIndex(index);
+  }
+  function DateListRow({ date, index }) {
+    let wert;
+    const currentDate = moment().format("YYYY-MM-DD");
+    const tomorrowDate = moment().add(1, "d").format("YYYY-MM-DD");
+    const thisDate = date.data.date;
+
+    thisDate === currentDate
+      ? (wert = "Heute")
+      : thisDate === tomorrowDate
+      ? (wert = "Morgen")
+      : moment(thisDate).week() === moment().week()
+      ? (wert = "Diese Woche")
+      : (wert = thisDate);
+
+    return (
+      <StyledDatesListRow $hover onClick={() => handelClickDate(index)}>
+        <p>{wert}</p>
+        <p>{date.data.promptlist}</p>
+        <FindPlace locationId={date.location} />
+      </StyledDatesListRow>
+    );
+  }
+
+  function FindPlace({ locationId }) {
+    return (
+      places.find((place) => place.id === locationId) && (
+        <>{places[locationId - 1].name}</>
+      )
+    );
+  }
+
+  function TranslateDateSeries({ form, frequency }) {
+    return form === "single" ? (
+      <p>Einzeltermin</p>
+    ) : (
+      <p>
+        Terminserie:
+        {frequency === "hour" && <>stündlich</>}
+        {frequency === "day" && <>täglich</>}
+        {frequency === "week" && <>wöchentlich</>}
+        {frequency === "month" && <>monatlich</>}
+        {frequency === "year" && <>järlich</>}
+      </p>
+    );
+  }
+
+  function CheckNoToday({ date, index }) {
+    if (index === 0 && date.data.date !== moment().format("YYYY-MM-DD"))
+      return (
+        <StyledDatesListRow>
+          <p>Heute nix</p>
+          <p>Chillen</p>
+          <p>Wo immer du magst</p>
+        </StyledDatesListRow>
+      );
+  }
+
+  return (
+    <>
+      {clickindex === null ? (
+        <StyeldDatesList>
+          <StyledDatesListRow $head>
+            <p key="wann">Wann</p>
+            <p key="was">Was</p>
+            <p key="wo">Wo</p>
+          </StyledDatesListRow>
+          {dates.length !== 0 ? (
+            sortDates.map((date, index) => (
```

Comment:
At the moment we are getting a warning in the console about keys not being unique. 

I think this is because we are potentially returning 2 things from this map. One idea to fix this would be to move the `CheckNoToday` outside of the loop. i.e. something like this. 

``` js
const noToday =
    sortDates[0] !== moment().format("YYYY-MM-DD") ? (
      <StyledDatesListRow>
        <p>Heute nix</p>
        <p>Chillen</p>
        <p>Wo immer du magst</p>
      </StyledDatesListRow>
    ) : null;

// function CheckNoToday({ date, index }) {
//   if (index === 0 && date.data.date !== moment().format("YYYY-MM-DD"))
//     return (
//       <StyledDatesListRow>
//         <p>Heute nix</p>
//         <p>Chillen</p>
//         <p>Wo immer du magst</p>
//       </StyledDatesListRow>
//     );
// }

```

and then in the loop do something like. 

``` js
{noToday}
{dates.length !== 0 ? (
  sortDates.map((date, index) => (
    <DateListRow key={date.id} date={date} index={index} />
  ))
) : (
  <StyledDatesListRow>Keine Termine</StyledDatesListRow>
)}
```

Reply:
Thank you for this suggestion. Works perfect.